src/test/ebpf_yaml.cpp

```
@@ -116,6 +116,42 @@ static EbpfProgramType make_program_type(const string& name, const ebpf_ctx_desc
+// analyze(EbpfDomain, ...)" — and analyze(EbpfDomain, ...) deliberately skips its own
+// clear so it doesn't orphan the cells we just allocated.
+static EbpfDomain string_invariant_to_entry_domain(const StringInvariant& inv, const bool setup_constraints,
+                                                   const AnalysisContext& context) {
+    ebpf_verifier_clear_thread_local_state();
+    return parse_string_invariant_to_domain(inv, setup_constraints, context);
+}
+
+// Bridge for observation-domain construction, called after analyze() has produced
+// invariants. Skips the thread-local clear because clearing here would destroy the
+// cell map entries that the result's invariants reference.
+static EbpfDomain string_invariant_to_observation_domain(const StringInvariant& inv, const bool setup_constraints,
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)